schema.c

```diff
@@ -4065,3 +4367,553 @@ sch_json_to_gnode (sch_instance * instance, sch_node * schema, json_t * json, in
+{
+    while (node)
+    {
+        if (g_strcmp0 (APTERYX_NAME (node), name) == 0)
+        {
+            *ret_node = node;
+            return true;
+        }
+        node = node->next;
+    }
+
+    *ret_node = NULL;
+    return false;
+}
+
+static bool
+xlat_tree_match (GNode *tree, GNode *pattern, GNode **last_match_node)
+{
+    bool match = false;
+    GNode *node1 = tree;
+    GNode *node2 = pattern;
+    GNode *ret_node;
+    GNode *last_valid_node = NULL;
+
+    while (node1 && node2)
+    {
+        match = true;
+        if (!sch_translate_sibling_exists (node1, APTERYX_NAME (node2), &ret_node) &&
+            g_strcmp0 ((char *) node2->data, "*") != 0)
+            return false;
+        if (ret_node)
+        {
+            last_valid_node = ret_node;
+            node1 = ret_node->children;
+            ret_node = NULL;
+        }
+        else
+        {
+            last_valid_node = node1;
+            node1 = node1->children;
+        }
+        node2 = node2->children;
+    }
+
+    if (node2)
+        match = false;
+    else
+        *last_match_node = last_valid_node;
+
+    return match;
+}
+
+static char *
+sch_translate_get_wildcard_data (GNode *node, GNode *tree, uint32_t pos)
+{
+    GNode *nnode = tree;
+    uint32_t matches = 0;
+
+    while (nnode && node)
+    {
+        if (g_strcmp0 (APTERYX_NAME (nnode), "*") == 0)
+        {
+            matches++;
+            if (matches == pos)
+                return g_strdup (APTERYX_NAME (node));
+        }
+
+        nnode = nnode->children;
+        node = node->children;
+    }
+    return NULL;
+}
+
+static inline gboolean
+_sch_translate_count_nodes (GNode *node, void *data)
+{
+    int *num_nodes = data;
+
+    if (node->data)
+        *num_nodes = *num_nodes + 1;
+
+    return false;
+}
+
+static inline gboolean
+_sch_translate_mark_children (GNode *node, gpointer data)
+{
+    GHashTable *node_table = data;
+    if (!g_hash_table_lookup (node_table, node))
+        g_hash_table_insert (node_table, node, node);
+
+    return false;
+}
+
+static void
+sch_translate_mark_children (GHashTable *node_table, GNode *node)
+{
+    if (node->children)
+    {
+        if (!g_hash_table_lookup (node_table, node))
+            g_node_traverse (node->children, G_PRE_ORDER, G_TRAVERSE_ALL, -1,
+                             _sch_translate_mark_children, node_table);
+    }
+}
+
+static void
+sch_translate_mark_node (GHashTable *node_table, GNode *node)
+{
+    GNode *parent;
+
+    if (!g_hash_table_lookup (node_table, node))
+    {
+        g_hash_table_insert (node_table, node, node);
+        parent = node->parent;
+        while (parent)
+        {
+            if (!g_hash_table_lookup (node_table, parent))
+                g_hash_table_insert (node_table, parent, parent);
+
+            parent = parent->parent;
+        }
+    }
+}
+
+static sch_node *
+sch_translate_schema (sch_instance * instance, GNode *node, int flags)
+{
```

Comment:
is this just a re-implementation of other functions?

Reply:
Removed routine as it was a duplicate of sch_traverse_get_schema